packages/expo-dev-menu/app/screens/MainScreen.tsx

```diff
@@ -0,0 +1,348 @@
+import {
+  View,
+  Text,
+  Heading,
+  Row,
+  Spacer,
+  XIcon,
+  Button,
+  Divider,
+  ClipboardIcon,
+  DebugIcon,
+  HomeFilledIcon,
+  InspectElementIcon,
+  PerformanceIcon,
+  RefreshIcon,
+  RunIcon,
+  StatusIndicator,
+  Image,
+} from 'expo-dev-client-components';
+import * as React from 'react';
+import { Switch } from 'react-native';
+
+import { useBuildInfo } from '../hooks/useBuildInfo';
+import { useClipboard } from '../hooks/useClipboard';
+import { useDevSettings } from '../hooks/useDevSettings';
+
+export function MainScreen() {
+  const buildInfo = useBuildInfo();
+  const { devSettings, actions } = useDevSettings();
+
+  const urlClipboard = useClipboard();
+  const buildInfoClipboard = useClipboard();
+
+  function onCopyUrlPress() {
+    const { hostUrl } = buildInfo;
+    urlClipboard.onCopyPress({ hostUrl });
+  }
+
+  function onCopyBuildInfoPress() {
+    const { runtimeVersion, sdkVersion, appName, appVersion } = buildInfo;
+    buildInfoClipboard.onCopyPress({ runtimeVersion, sdkVersion, appName, appVersion });
+  }
+
+  const hasCopiedBuildInfoContent = Boolean(buildInfoClipboard.clipboardContent);
+
+  return (
+    <View flex="1" bg="secondary">
+      <View padding="medium" bg="default">
+        <Row align="start">
+          <Row align="center">
+            <View>
+              <View height="xl" width="xl" overflow="hidden" bg="secondary" rounded="medium">
+                {Boolean(buildInfo.appIcon) && (
+                  <Image
+                    source={{ uri: buildInfo.appIcon }}
+                    style={{ flex: 1, resizeMode: 'contain' }}
+                  />
+                )}
+              </View>
+            </View>
+
+            <Spacer.Horizontal size="small" />
+
+            <View>
+              <Heading size="small" weight="bold">
+                {buildInfo.appName}
+              </Heading>
+              <Spacer.Vertical size="tiny" />
+              {Boolean(buildInfo.runtimeVersion) && (
+                <>
+                  <Text size="small" color="secondary">
+                    {`Runtime version: ${buildInfo.runtimeVersion}`}
+                  </Text>
+                </>
+              )}
+
+              {Boolean(buildInfo.sdkVersion) && !buildInfo.runtimeVersion && (
+                <>
+                  <Text size="small" color="secondary">
+                    {`SDK version: ${buildInfo.sdkVersion}`}
+                  </Text>
+                </>
+              )}
+            </View>
+          </Row>
+
+          <Spacer.Horizontal size="flex" />
+
+          <Button.ScaleOnPressContainer
+            bg="ghost"
+            rounded="full"
+            minScale={0.8}
+            onPress={actions.closeMenu}>
+            <View padding="micro">
+              <XIcon />
+            </View>
+          </Button.ScaleOnPressContainer>
+        </Row>
+      </View>
+
+      <Divider />
+
+      {Boolean(buildInfo.hostUrl) && (
```

Comment:
this is probably fine for now, but for the longer term i'm not sure this is the right logic here -- it seems like if we're viewing a published update we should be showing the URL to the update (manifest) here, whether that's `hostUrl` or a different field value

Reply:
yes I am meaning to follow up with you about getting the right value here!

Reply:
I think the correct value would probably be DevLauncherController.manifestURL (or derived from it), may need to find a way to pass that value into here though.